apps/vscode-extension/src/commands/extract/styling/helpers/utils.helper.ts

```diff
@@ -0,0 +1,195 @@
+import { Position, Selection, TextDocument, TextEditorEdit, TextLine, workspace } from 'vscode';
+import { basename } from 'node:path';
+import { EOL } from 'node:os';
+import {regExp} from './regex.helper';
+import * as vscode from 'vscode';
+
+/**
+ * Sanitaze the selector
+ *
+ * @param selector to clean
+ * @returns the cleaned selector
+ */
+function cleanSelector(selector: string): string {
+  return selector.replaceAll(regExp.trimLeft, '')
+    .replaceAll(regExp.trimRight, '')
+    .replaceAll(regExp.sanitizer, '-');
+}
+
+/**
+ * Generate the proposed variable name based of the path of selectors.
+ * Also make sure to not include any duplicated selectors or any forbidden words from the config `forbiddenWords`
+ *
+ * @param selectors to used to generate the name
+ * @param fileName name of the file to append to the generated name
+ * @returns the final proposed name
+ */
+function generateScssVarName(selectors: string[], fileName: string): string {
+  let name = '';
+  const forbiddenWords = vscode.workspace.getConfiguration('otter.extract.styling').get<string>('forbiddenWords');
+  const wordsSet = new Set<string>([...forbiddenWords ?? [], '']);
+  const prefixToAdd = workspace.getConfiguration('otter.extract.styling').get<string>('prefix');
+
+  selectors = [...([prefixToAdd ?? '']), fileName, ...selectors];
+  selectors.forEach((element) => {
+    const elementWords = element.split('-');
+    let distinctSelector = '';
+    elementWords.forEach((word) => {
+      if (!wordsSet.has(word)) {
+        wordsSet.add(word);
+        distinctSelector = distinctSelector.concat(word, '-');
+      }
+    });
+    name = name.concat(distinctSelector);
+  });
+  return name;
+}
+
+/**
+ * Go through the tree of selectors and keep only the path to the selected property
+ *
+ * @param css
+ * @param selectors
+ * @returns list of used selectors
+ */
+function keepOnlyUsedSelector(css: string, selectors: string[]): string[] {
+  if (!css || !selectors) {
```

Comment:
It is a bit unclear that you want to keep the selectors that are not in a closing block (since your css block will not contain anything after your selected property). Do not hesitate to add some comments to help future debugs/enhancements :)